src/System.Private.ServiceModel/tests/Common/Infrastructure/xunit/ConditionAttribute.cs

```diff
@@ -0,0 +1,121 @@
+
+            List<string> falseConditions = new List<string>(conditionCount);
+
+            foreach (string entry in Conditions)
+            {
+                string conditionMemberName = entry;
+
+                // Null condition member names are silently tolerated
+                if (string.IsNullOrWhiteSpace(conditionMemberName))
+                {
+                    continue;
+                }
+
+                string[] symbols = conditionMemberName.Split('.');
+                Type declaringType = testMethodDeclaringType;
+
+                if (symbols.Length == 2)
+                {
+                    conditionMemberName = symbols[1];
+                    ITypeInfo type = testMethod.TestClass.Class.Assembly.GetTypes(false).Where(t => t.Name.Contains(symbols[0])).FirstOrDefault();
+                    if (type != null)
+                    {
+                        declaringType = type.ToRuntimeType();
+                    }
+                }
+
+                MethodInfo conditionMethodInfo;
+                if ((conditionMethodInfo = LookupConditionalMethod(declaringType, conditionMemberName)) == null)
+                {
+                    falseConditions.Add(String.Format("Condition \"{0}\" not found in type \"{1}\".",
+                                                      conditionMemberName, testMethodDeclaringType.Name));
+                    continue;
+                }
+
+                // In the case of multiple conditions, collect the results of all
+                // of them to produce a summary skip reason.
+                try
+                {
+                    if (!(bool)conditionMethodInfo.Invoke(null, null))
+                    {
+                        falseConditions.Add(conditionMemberName);
+                    }
+                }
+                catch (Exception exc)
+                {
+                    falseConditions.Add(String.Format("Condition \"{0}\" threw exception \"{1}\".",
+                                                      conditionMemberName, exc.ToString()));
+                }
+            }
+
+            if (falseConditions.Count == 0)
+            {
+                return null;
+            }
+
+            return String.Format("{0}", String.Join(",", falseConditions));
+        }
+
+
+        internal static MethodInfo LookupConditionalMethod(Type t, string name)
+        {
+            if (t == null || name == null)
+                return null;
+
+            TypeInfo ti = t.GetTypeInfo();
+
+            MethodInfo mi = ti.GetDeclaredMethod(name);
+            if (mi != null && mi.IsStatic && mi.GetParameters().Length == 0 && mi.ReturnType == typeof(bool))
+                return mi;
+
+            PropertyInfo pi = ti.GetDeclaredProperty(name);
+            if (pi != null && pi.PropertyType == typeof(bool) && pi.GetMethod != null && pi.GetMethod.IsStatic && pi.GetMethod.GetParameters().Length == 0)
```

Comment:
>pi.GetMethod.GetParameters() [](start = 114, length = 28)

Is it possible for a PropertyInfo to return a method from GetMethod which takes any parameters? Also, PropertyInfo.GetMethod will return non-public methods too. Is this what we want? This might cause us problems once .Net native is using this.